src/libgt2/libgt2.hpp

```diff
@@ -0,0 +1,80 @@
+//*****************************************************************************
+// Copyright 2025 Intel Corporation
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//*****************************************************************************
+#ifndef SRC_LIBGT2_LIBGT2_HPP_
+#define SRC_LIBGT2_LIBGT2_HPP_
+#include <string>
+#include <memory>
+
+#include <assert.h>
+#include <fcntl.h>
+#include <git2.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+#include <sys/stat.h>
+#include <sys/types.h>
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+namespace ovms {
+class Status;
+
+class Libgt2InitGuard {
+public:
+    int status;
+    std::string errMsg;
+    Libgt2InitGuard() {
+        this->status = git_libgit2_init();
+        if (this->status < 0) {
+            const git_error* err = git_error_last();
+            const char* msg = err ? err->message : "unknown failure";
+            errMsg = std::string(msg);
+        } else {
+            errMsg = "";
+        }
+    }
+    ~Libgt2InitGuard() {
+        git_libgit2_shutdown();
+    }
+};
+
+class HfDownloader {
+public:
+    HfDownloader(const std::string& sourceModel, const std::string& downloadPath, bool pullHfModelMode, const std::string& hfEndpoint, const std::string& hfToken, const std::string& httpProxy);
+    Status cloneRepository();
+    bool isPullHfModelModeOn();
+
+protected:
+    std::string sourceModel;
+    std::string downloadPath;
+    std::string hfEndpoint;
+    std::string hfToken;
+    std::string httpProxy;
+    bool pullHfModelMode;
```

Comment:
Why is this here? Either we create module for pulling or not. I don't see a point.

Reply:
Left because from previous implementation. Good catch to remove.